apps/gateway/lib/gateway/controllers/auth_controller.ex

```diff
@@ -36,7 +36,7 @@ defmodule Gateway.Controllers.AuthController do
     with {:ok, claims} <- TokenManager.verify(gateway_jwt),
          hashed_client_id = :crypto.hash(:sha256, client_id),
          {:ok, ^hashed_client_id} <- Base.url_decode64(claims["dev"]),
-         {:ok, user} <- Users.get_user(claims["sub"]) do
+         {:ok, %{user: user}} <- Users.get_user_and_maybe_insert_daily_quests(claims["sub"]) do
```

Comment:
Can't we do this separately? I think it would add more clarity and wouldn't replace former get_user behavior.
Just another with clause to "maybe_insert_daily_quests".

Reply:
Sure thing, i'll split the quest generation and user retrieval in this module